lib/gollum/public/gollum/javascript/editor/langs/markdown.js

```diff
@@ -6,8 +6,35 @@
 
 (function($) {
 
-var MarkDown = $.extend(true, {}, $.DefaultLang); // The default language definition is markdown, so copy it into a new variable.
-$.GollumEditor.defineLanguage('markdown', MarkDown);
+// The default language definition (in default.js) is markdown, so we only need to set the special helper functions.
+var MarkDown = {
+
+  'gollum-helpers': {
+
+    'find-header-line': function (line, token) {
+      // For use in sections.js
+      // Checks if token matches alternative-syntax markdown headers. These are of the form '===' or '---'.
+      // If so, returns an array containing the text of the header and the previous line number.
+      // If not, returns null.
+      if ( token.match(/(^[=]+$)/) || token.match(/(^[-]+$)/) ){
```

Comment:
I would prefer to use just one regex. Something like this might do the trick:
```regex
^([=]{3,}|[-]{3,})$
```
(Three or more equal signs or three or more dashes, going from the beginning of the line to the end of the line.)

Reply:
👍